SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -351,6 +358,46 @@ bool FLocalDeploymentManager::SetupRuntimeFileLogger(const FString& RuntimeLogDi
 }
 
 bool FLocalDeploymentManager::TryStopLocalDeployment()
+{
+	if (!StartLocalDeploymentShutdown())
+	{
+		return false;
+	}
+
+	RuntimeProcess->Stop();
+
+	return FinishLocalDeploymentShutdown();
+}
+
+bool FLocalDeploymentManager::TryStopLocalDeploymentGracefully()
+{
+	if (bTestRunnning)
+	{
+		bTestRunnning = false;
+		return TryStopLocalDeployment();
+	}
+
+	if (!StartLocalDeploymentShutdown())
+	{
+		return false;
+	}
+
+	FHttpModule& HttpModule = FModuleManager::LoadModuleChecked<FHttpModule>("HTTP");
+#if ENGINE_MINOR_VERSION >= 26
+	TSharedRef<IHttpRequest, ESPMode::ThreadSafe> HttpRequest = HttpModule.Get().CreateRequest();
+#else
+	TSharedRef<IHttpRequest> HttpRequest = HttpModule.Get().CreateRequest();
+#endif
+
+	HttpRequest->SetURL(TEXT("http://localhost:5006/shutdown"));
+	HttpRequest->SetVerb("GET");
+
+	HttpRequest->ProcessRequest();
+
+	return FinishLocalDeploymentShutdown();
```

Comment:
If the runtime fails to gracefully shutdown it can be left running - maybe a fallback to terminate the process after a certain amount of time?